packages/kbn-grid-layout/grid/grid_row.tsx

```diff
@@ -40,7 +41,7 @@ export const GridRow = forwardRef<
     gridRef
   ) => {
     const currentRow = gridLayoutStateManager.gridLayout$.value[rowIndex];
-    const [panelIds, setPanelIds] = useState<string[]>(Object.keys(currentRow.panels));
+    const [panelIds, setPanelIds] = useState<string[]>(getKeysInOrder(currentRow.panels));
```

Comment:
We aren't responding to the grid layout being updated:


https://github.com/user-attachments/assets/22eff635-928e-4106-9674-963c2eee1412

But I'm cautious about this 🙈 The problem is, if we update the HTML ordering in response to `gridLayoutStateManager.gridLayout` changes, we will be re-rendering things **a lot** as the user drags - and this essentially removes most of the benefits of doing all of our styles via subscription. 

We could **help** with this by only rearranging the HTML on drop? Maybe we try this and do some performance measuring to see how that goes 🤔 Interested in your thoughts @mbondyra

Reply:
You're totally right! I only checked adding and removing panels and thought we're done 😬 

I added an `onInteractionComplete` callback to update the ordered `panelIds` after a drop event. This could be further optimized by first comparing the current ordered `panelIds` with a sorted version from the state management and only triggering updates if there's a difference. 🤔

But we definitely cannot ignore it, it's not just about focus management, but also how screen readers read the screen. The `tabindex` idea is not a viable alternative as we would have to set tabindex to every element on the page, if we start with adding numbered `tabindex` (!==0 or !==-1).

The PR is not finished yet, I started with the tests and got stuck in the middle 🙈